website/blog/2024-05-14-monitor-ferretdb-performance-using-coroot.md

```diff
@@ -0,0 +1,180 @@
+    volumes:
+      - coroot_data:/data
+    ports:
+      - 8080:8080
+    command:
+      - '--bootstrap-prometheus-url=http://prometheus:9090'
+      - '--bootstrap-refresh-interval=15s'
+      - '--bootstrap-clickhouse-address=clickhouse:9000'
+    depends_on:
+      - clickhouse
+      - prometheus
+
+  node-agent:
+    image: ghcr.io/coroot/coroot-node-agent
+    privileged: true
+    pid: 'host'
+    volumes:
+      - /sys/kernel/debug:/sys/kernel/debug
+      - /sys/fs/cgroup:/host/sys/fs/cgroup
+    command:
+      - '--collector-endpoint=http://coroot:8080'
+      - '--cgroupfs-root=/host/sys/fs/cgroup'
+
+  prometheus:
+    image: prom/prometheus:v2.45.4
+    volumes:
+      - prometheus_data:/prometheus
+    command:
+      - '--config.file=/etc/prometheus/prometheus.yml'
+      - '--storage.tsdb.path=/prometheus'
+      - '--web.console.libraries=/usr/share/prometheus/console_libraries'
+      - '--web.console.templates=/usr/share/prometheus/consoles'
+      - '--web.enable-lifecycle'
+      - '--web.enable-remote-write-receiver'
+    ports:
+      - '127.0.0.1:9090:9090'
+
+  clickhouse:
+    image: clickhouse/clickhouse-server:24.3
+    ports:
+      - '127.0.0.1:9000:9000'
+    ulimits:
+      nofile:
+        soft: 262144
+        hard: 262144
+
+  postgres:
+    image: postgres
+    environment:
+      - POSTGRES_USER=username
+      - POSTGRES_PASSWORD=password
+      - POSTGRES_DB=ferretdb
+    volumes:
+      - ./data:/var/lib/postgresql/data
+
+  ferretdb:
+    image: ghcr.io/ferretdb/ferretdb
+    restart: on-failure
+    ports:
+      - 27017:27017
+    environment:
+      - FERRETDB_POSTGRESQL_URL=postgres://postgres:5432/ferretdb
+```
+
+The coroot-node-agent is a Prometheus exporter for gathering container metrics running on a particular node.
+It gathers log-based metrics, network latency, JVM metrics, and more.
+
+Coroot offers a distributed tracing approach that allows engineers to visualize request paths across all components.
+This could help to quickly identify latency issues, errors, or bottlenecks in their setup.
+
+Depending on your application setup, you may need to modify the Docker compose `yaml` file and configure the tracing metrics on Coroot by enabling OpenTelemetry locally.
+[Check this docs to learn more](https://coroot.com/docs/coroot-community-edition/tracing/overview).
+
+Once you apply and run the `docker compose up`, Docker starts and runs the entire application defined in the `yaml` file.
+
+Run `docker compose ps` just to be sure it's correctly deployed.
+Since Coroot was deployed locally, you can access it at http://localhost:8080/.
+
+## Monitor FerretDB performance with Coroot
+
+The Coroot dashboard provides the full details on all components.
+
+At first glance, we can see a memory leak on the `ferretdb` and `postgres` databases.
+That suggests that allocated memory is not being efficiently reused or deallocated, causing the total memory usage to grow progressively as the services operate.
```

Comment:
… so we want to publish a blog post that casually mentions that FerretDB and PostgreSQL have memory leaks just like that?